### ISSUE TITLE
[MAF] Variant color coding mismatch

### ISSUE THREAD

**Issue by tilschaef** [sbslee/fuc#67]
First, thanks for providing us with this awesome tool! 😊

I recently noticed some strange behavior when creating a custom oncoplot. Specifically, the coloring of the variant classification in the gene plot did not match the variant color coding in the waterfall plot when count was set to 1.  

`data_mf.plot_waterfall(count=1)`

I eventually resolved it by setting `vmin `and` vmax `to the maximum/minimum index of `NONSYN_COLORS` as the anchors were not correctly inferred by `seaborn.heatmap` in `plot_waterfall  `

**Comment by sbslee** [comment]
@tilschaef,

Thanks so much for reporting this bug! I can confirm that the issue seems to be specific to when `count=1` because it behaves correctly when, say, `count=2`:

```
import matplotlib.pyplot as plt
from fuc import common, pymaf
common.load_dataset('tcga-laml')
maf_file = '~/fuc-data/tcga-laml/tcga_laml.maf.gz'
mf = pymaf.MafFrame.from_file(maf_file)
fig, axes = plt.subplots(3, 2, figsize=(14, 8))
mf.plot_waterfall(ax=axes[0, 0], linewidths=0.5)
mf.plot_waterfall(ax=axes[0, 1], count=1, linewidths=0.5)
mf.plot_waterfall(ax=axes[1, 0], count=2, linewidths=0.5)
mf.plot_waterfall(ax=axes[1, 1], count=3, linewidths=0.5)
mf.plot_waterfall(ax=axes[2, 0], count=4, linewidths=0.5)
mf.plot_waterfall(ax=axes[2, 1], count=5, linewidths=0.5)
axes[0, 0].set_title('Default')
axes[0, 1].set_title('count=1')
axes[1, 0].set_title('count=2')
axes[1, 1].set_title('count=3')
axes[2, 0].set_title('count=4')
axes[2, 1].set_title('count=5')
plt.tight_layout()
``` 

![bug](https://user-images.githubusercontent.com/21111749/227062181-d22d7309-553f-4363-9d42-015d439b92ff.png)

I will look into it ASAP and get back to you with a solution.

**Comment by sbslee** [comment]
@tilschaef,

The bug has been fixed in the `0.37.0-dev` branch. If you use the same code you will get:

![bug](https://user-images.githubusercontent.com/21111749/227068285-8a5a01a4-99ad-4127-bea7-1d80a73fc255.png)

As you suggested, it was a single-line fix by specifying `vmin` and `vmax` to the `seaborn.heatmap` method so that it consistently use the same color scale instead of letting it infer the scale itself from the input data. FYI, I used `vmax=len(colors)` instead of `vmax=len(NONSYN_COLORS)`.

Hope this helps. Let me know if you have further questions!
